src/Compilers/CSharp/Test/Semantic/Semantics/PrimaryConstructorTests.cs

```diff
@@ -11699,6 +11742,1122 @@ class Base
+                // (7,36): error CS0212: You can only take the address of an unfixed expression inside of a fixed statement initializer
+                // unsafe class C1(int x, S s) : Base(&x, &s.f)
+                Diagnostic(ErrorCode.ERR_FixedNeeded, "&x").WithLocation(7, 36),
+                // (7,40): error CS0212: You can only take the address of an unfixed expression inside of a fixed statement initializer
+                // unsafe class C1(int x, S s) : Base(&x, &s.f)
+                Diagnostic(ErrorCode.ERR_FixedNeeded, "&s.f").WithLocation(7, 40)
+                );
+        }
+
+        [Fact]
+        public void ParameterCapturing_094_DefiniteAssignment()
+        {
+            var text = @"
+class C1(int x, S s, string y)
+{
+    void M1()
+    {
+        _ = x + s.f + y.Length;
+    }
+
+    void M2()
+    {
+        _ = s.f + y.Length + x;
+    }
+
+    void M3()
+    {
+        _ = y.Length + x + s.f;
+    }
+
+    void M4()
+    {
+        _ = x;
+    }
+
+    void M5()
+    {
+        _ = s.f;
+    }
+
+    void M6()
+    {
+        _ = y.Length;
+    }
+}
+
+struct S
+{
+    public int f;    
+    public S(int x) => f = x; 
+}
+";
+            CreateCompilation(text).VerifyEmitDiagnostics();
+        }
+
+        [Fact]
+        public void ParameterCapturing_095_DefiniteAssignment()
+        {
+            var text = @"
+class C1(out int x, out S s, out string y)
+{
+    void M1()
+    {
+        _ = x + s.f + y.Length;
+    }
+
+    void M2()
+    {
+        _ = s.f + y.Length + x;
+    }
+
+    void M3()
+    {
+        _ = y.Length + x + s.f;
+    }
+
+    void M4()
+    {
+        _ = x;
+    }
+
+    void M5()
+    {
+        _ = s.f;
+    }
+
+    void M6()
+    {
+        _ = y.Length;
+    }
+}
+
+struct S
+{
+    public int f;    
+    public S(int x) => f = x; 
+}
+";
+            CreateCompilation(text).VerifyEmitDiagnostics(
+                // (2,7): error CS0177: The out parameter 'x' must be assigned to before control leaves the current method
+                // class C1(out int x, out S s, out string y)
+                Diagnostic(ErrorCode.ERR_ParamUnassigned, "C1").WithArguments("x").WithLocation(2, 7),
+                // (2,7): error CS0177: The out parameter 's' must be assigned to before control leaves the current method
+                // class C1(out int x, out S s, out string y)
+                Diagnostic(ErrorCode.ERR_ParamUnassigned, "C1").WithArguments("s").WithLocation(2, 7),
+                // (2,7): error CS0177: The out parameter 'y' must be assigned to before control leaves the current method
+                // class C1(out int x, out S s, out string y)
+                Diagnostic(ErrorCode.ERR_ParamUnassigned, "C1").WithArguments("y").WithLocation(2, 7),
+                // (6,13): error CS1628: Cannot use ref, out, or in parameter 'x' inside an anonymous method, lambda expression, query expression, or local function
```

Comment:
> The message seems incorrect since there is no lambda or local function involved. Should the message be updated or a distinct error reported?

This is tracked by a PROTOTYPE comment  on line 13036.